specification/securityinsights/resource-manager/Microsoft.SecurityInsights/stable/2020-01-01/SecurityInsights.json

```diff
@@ -1030,378 +2045,529 @@
+      "type": "object"
+    },
+    "FusionAlertRule": {
+      "allOf": [
+        {
+          "$ref": "#/definitions/AlertRule"
+        }
       ],
-      "type": "string",
-      "x-ms-enum": {
-        "modelAsString": true,
-        "name": "AlertSeverity",
-        "values": [
-          {
-            "description": "High severity",
-            "value": "High"
-          },
-          {
-            "description": "Medium severity",
-            "value": "Medium"
-          },
-          {
-            "description": "Low severity",
-            "value": "Low"
+      "description": "Represents Fusion alert rule.",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/FusionAlertRuleProperties",
+          "description": "Fusion alert rule properties",
+          "x-ms-client-flatten": true
+        }
+      },
+      "type": "object",
+      "x-ms-discriminator-value": "Fusion"
+    },
+    "FusionAlertRuleProperties": {
+      "description": "Fusion alert rule base property bag.",
+      "properties": {
+        "alertRuleTemplateName": {
+          "description": "The Name of the alert rule template used to create this rule.",
+          "type": "string"
+        },
+        "description": {
+          "description": "The description of the alert rule.",
+          "readOnly": true,
+          "type": "string"
+        },
+        "displayName": {
+          "description": "The display name for alerts created by this alert rule.",
+          "readOnly": true,
+          "type": "string"
+        },
+        "enabled": {
+          "description": "Determines whether this alert rule is enabled or disabled.",
+          "type": "boolean"
+        },
+        "lastModifiedUtc": {
+          "description": "The last time that this alert has been modified.",
+          "format": "date-time",
+          "readOnly": true,
+          "type": "string"
+        },
+        "severity": {
+          "$ref": "#/definitions/AlertSeverity",
+          "description": "The severity for alerts created by this alert rule.",
+          "readOnly": true
+        },
+        "tactics": {
+          "description": "The tactics of the alert rule",
+          "items": {
+            "$ref": "#/definitions/AttackTactic"
           },
-          {
-            "description": "Informational severity",
-            "value": "Informational"
-          }
-        ]
-      }
+          "readOnly": true,
+          "type": "array"
+        }
+      },
+      "required": [
+        "alertRuleTemplateName",
+        "enabled"
+      ],
+      "type": "object"
     },
-    "AlertsDataTypeOfDataConnector": {
-      "description": "Alerts data type for data connectors.",
+    "FusionAlertRuleTemplate": {
+      "allOf": [
+        {
+          "$ref": "#/definitions/AlertRuleTemplate"
+        }
+      ],
+      "description": "Represents Fusion alert rule template.",
       "properties": {
-        "alerts": {
+        "properties": {
```

Comment:
do not use an allof for properties, use a ref to the defining object

Reply:
done

Reply:
Apparently, this is a breaking change - the breaking change check is failing after that change
is it fine @markcowl ?

Reply:
That's unfortunate.  It should be equivalent, but it's OK to switch back to avoid the breaking change

Reply:
I am afraid we shouldn't switch back
There is a problem in the previous definition with the nested properties definition, which was not correct. Switching back to the allOf in the property forces us to return the nested properties field.

Shall we keep it like that.. I think it is not the only Breaking Change anyway.. The CloudError that you suggested to change (to use the common one) is also a breaking change

Reply:
We should make sure the swagger spec is correct, so if the previous definition was incorrect, we should correct it.

On CloudError, this is a usability problem with the errors, but if it would actually be a breaking change, then we should not make the breaking change.